kedro/io/cached_dataset.py

```diff
@@ -50,6 +51,7 @@ def __init__(
             copy_mode: The copy mode used to copy the data. Possible
                 values are: "deepcopy", "copy" and "assign". If not
                 provided, it is inferred based on the data type.
+            metadata: Any arbitrary user metadata.
```

Comment:
I left a similar comment on `kedro-plugins`, but I'd add a bit more info here about metadata not being used within Kedro.